in-progress/8131-forced-inclusion.md

```diff
@@ -0,0 +1,525 @@
+If we did not have this check, it would be possible to provide a proof for a transactions that was included in a previous epoch.
+This could happen for example if you also had a block in the prior epoch.
+Lets say that there are 2 transactions that have passed their "deadline" so they must be included.
+You provide a block with 2 transactions, the first is from the forced queue, but the second is not.
+The `force_until` would return that you need to progress by 2.
+You then provide it the proof for the prior tx, it progress by 1, and then the proof for your new, the second.
+At this point, you progressed by 2, but only one of them was actually new.
+
+If someone have included a forced transaction earlier than it was required, anyone can use the `show_included` to mark it as included and progress the state of the delayed queue.
+
+Notice that the forced inclusions outlined does **not** strictly require insertion into the beginning of the epoch.
+But only that it is within the start of blocks within the epoch.
+
+Also, observe, that a failure to include the forced transactions will be caught at the time of proof verification, not at the time of proposal!
+
+#### Stricter inclusion constraints
+
+By being stricter with our constraints, and more loose with our gas expenditure 😎 we can change the system such that it:
+- Enforces the ordering earlier in the life-cycle;
+- Force the delayed queue to be included as the first transactions, e.g., from the very start of the epoch, limiting the proposers power.
+
+By extending the checks of the proposal, we can cause an early failure, such that we don't have to wait until the block have been proven to catch them meddling with the force inclusion ordering.
+
+It requires a bit more accounting, and will likely be slightly more expensive, however, the improved properties seems to make up for that.
+
+Could I not have written this in the first go?
+Well, yes, but it might get confusing with both the pending and non pending functions so I'm going to keep them separate for now.
+
+
+```python
+# Tracks the current active "reality"
+active_pending: uint256
+# For all the potential realities, tracks where the tip is
+forced_pending_tip: HashMap[uint256, uint256]
+# For all the potential realities, tracks what have been included
+included_pending: HashMap[uint256, HashMap[uint256, bool]]
+
+def propose(header: Header, fips: ForceInclusionProof[]):
+  super.propose(header)
+
+  epoch = self.get_epoch_at(header.global_variables.timestamp)
+  force_until = force_until_pending(epoch, header.tx_count)
+
+  for i, fip in enumerate(fips):
+    assert fip.block_number == header.global_variables.block_number, 'incorrect block'
+    assert fip.index_in_block == i, 'incorrect ordering'
+    assert self.show_included_pending(fip), 'did not progress the queue, incorrect ordering'
+
+  assert forced_until <= self.forced_pending_tip[self.active_pending], 'missing force inclusions'
+
```

Comment:
Added a comment for it 
```
> Consideration around minimum block size:  
> Since one attack vector a censor could employ is to create blocks of just 1 transaction from the forced queue and thereby censor most for a longer duration it might be beneficial to require a minimum block size.
```